### PR TITLE
Test and fix corner-cases of Type.of and Type.CtorN.of on Scala 2 and Scala 3

### PR DESCRIPTION
**Pull request by MateuszKubuszok**
- [x] #128
  - [x] add tests
  - [x] update Scala 2 implementation to fix the bug
- [x] #126
  - [x] add tests
  - [x] update Scala 2 implementation to fix the bug
  - [x] update Scala 3 implementation to fix the bug
- [x] update documentation to explain what is supported and what is not

Kind-projectors are deferred to #139 

---

> [!NOTE]
> Strengthens Cross-Quotes Type.of and Type.CtorN behavior on Scala 2/3, adds comprehensive tests up to Ctor22, and documents limitations and logging.
> 
> - **Cross-Quotes (Scala 2 macro)**:
>   - Implement `TypeWithUnderlyingInjected` to correctly resolve `Type.of[...]` with local/imported implicits and `Underlying` in `WeakTypeTag`-constrained contexts.
>   - Rework `Type.CtorN.of` expansions (N=1–22): use `Type.CtorN.Apply/Stub` helpers, better unapply matching, and safer typechecked code paths.
>   - Improve logging control and message emission; include line/column; minor fresh-name/nowarn and safety tweaks.
> - **Cross-Quotes (Scala 3 plugin)**:
>   - Expand logging filter to file|line|column and emit through reporter.
>   - Inject local givens for `Type[A]` from context bounds/implicit params/imported `Underlying`; add detection for `Type[A]`-returning givens/defs.
> - **Types API (Scala 2)**:
>   - Add `Type.CtorN.Apply`/`Stub` type aliases to support macro expansions.
> - **Printers**:
>   - Better handling/printing of `TypeTree` and `Expr` blocks; minor comment/style fixes.
> - **Tests**:
>   - Add exhaustive fixtures and specs for `Type.of` resolution and `Type.Ctor1..22` (including aliases/fixed params); introduce `examples/kinds` arity and alias types.
>   - Streamline old fixtures (remove/simple type cases) and add `testTypeOf`.
> - **Docs**:
>   - Major update to Cross-Quotes limitations and behavior on Scala 2/3, including workarounds and plugin constraints.
>   - Minor doc/code sample cleanups.
> - **Settings/Config**:
>   - Enhance logging setting parsing (supports `file|line|column`); clarify `log.cross-quotes` value format in docs/dev properties.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 904a34b3ea7dbf0f9038cee5740217613a7fcb56. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: hearth-tests/src/main/scala/hearth/typed/CrossTypesFixturesImpl.scala [hunk 1]
```diff
@@ -6,108 +6,505 @@ import hearth.data.Data
...
+        //  - declared val A: Type[A] = someType (declared locally or inherited)
+        // all would be considered when building a type that uses A.
+        "resolvingImplicitType" -> Data.map(
+          "viaTypeBound" -> Data {
+            Type.of[Option[A]].plainPrint
+          },
+          "viaImport" -> Data {
+            val b = Type[A].as_??
+            import b.Underlying as B
+            Type.of[Option[B]].plainPrint // compiles on Scala 2 but with the wrong result (implicit is ignored, and compiler creates Type[b.Underlying])
```
Comment (MateuszKubuszok) [inline hearth-tests/src/main/scala/hearth/typed/CrossTypesFixturesImpl.scala]: Scala 2 `c.WeakTypeTag` implementation has a limitation, that it does **not** look at implicit `c.WeakTypeTag`s unless the code is inside `def method[A](implicit A: c.WeakTypeTag[A])` a method with a type parameter and implicit tag for it.

We would have to detect all such cases and explicitly wrap into:

```scala
def workaround[A: c.WeakTypeTag, B: c.WeakTypeTag] = ...
workaround(A.asInstanceOf[c.WeakTypeTag[A]], B.asInstanceOf[c.WeakTypeTag[B]])
```

to work around it. The issue is: how to reliably detect which types require a workaround?

File: hearth-tests/src/main/scala/hearth/typed/CrossTypesFixturesImpl.scala [hunk 2]
```diff
@@ -6,108 +6,505 @@ import hearth.data.Data
...
+            body(using Ctx(Type[A]))
+          },
+          "viaInheritance" -> Data {
+            val a = Type[A]
+            trait HelperA {
+              type B
+              implicit val B: Type[B]
+            }
+            trait HelperB extends HelperA {
+              def result = Type.of[Option[B]].plainPrint // fails to compile on Scala 3 (we aren't picking up the implicit Type[B])
```
Comment (MateuszKubuszok) [inline hearth-tests/src/main/scala/hearth/typed/CrossTypesFixturesImpl.scala]: Same issue as above. Here the additional compilcation is that plugin works on untyped trees, so we cannot ask the compiler about implicits, which can be defined in some other scope.

File: hearth-tests/src/main/scala/hearth/typed/CrossTypesFixturesImpl.scala (hunk 1 above)
Comment (MateuszKubuszok) [reply hearth-tests/src/main/scala/hearth/typed/CrossTypesFixturesImpl.scala]: Work in progress, as of [399b7fa](https://github.com/MateuszKubuszok/hearth/pull/136/commits/399b7faf63066a86e71f87d1ca2fb0252fbbb762) we are creating the `def workaround` and detecting correctly what needs a workaround... but the body needs to be rewritten: code sees e.g. `b.Underlying` type instead of `B`, so the injected implicits are ignored. We have to transform the body as well.
